src/game/gamecore.cpp

```diff
@@ -376,7 +376,7 @@ void CCharacterCore::Move()
 				if(!pCharCore || pCharCore == this)
 					continue;
 				float D = distance(Pos, pCharCore->m_Pos);
-				if(D < PhysSize && D > 0.0f)
+				if((D < PhysSize && D > 0.0f) || (D <= 0.001f && D >= -0.001f))
```

Comment:
Hm, that looks odd, `D` shouldn't be negative at all.

Reply:
I took the code that DDNet had in for several years, it works reliably.

Reply:
https://github.com/ddnet/ddnet/blob/af6d463fb6ed356e7e5bdede15fa54646284f17c/src/game/gamecore.cpp#L505

Reply:
This would imply that sqrtf can return a negative value: 
```cpp
template<typename T>
inline T distance(const vector2_base<T> &a, const vector2_base<T> &b)
{
	return length(a-b);
}
inline float length(const vector2_base<float> &a)
{
	return sqrtf(a.x*a.x + a.y*a.y);
}
```

Reply:
shall i remove the negative check?

Reply:
Okay, removed the negative check

Reply:
There could be a negative zero as result.